editoast/src/views/projection.rs

```diff
@@ -290,96 +269,108 @@ pub async fn compute_projected_train_paths(
         conn,
         valkey_client.clone(),
         core_client.clone(),
-        &trains_schedules,
+        &train_schedules,
         infra,
         electrical_profile_set_id,
     )
     .await?;
 
     // 2. Extracts train simulation details and computes unique hashes for projected train paths.
-    let trains_details = extract_train_details(&trains_schedules, simulations).await?;
+    let trains_details = extract_train_details(simulations).await?;
 
-    let mut trains_hash_values = vec![];
+    let mut trains_hash_values: HashMap<String, Vec<usize>> = HashMap::new();
 
-    trains_hash_values.extend(trains_details.iter().map(|t| {
-        t.compute_projection_hash_with_versioning(infra.id, infra.version, &track_section_ranges)
-    }));
+    trains_details
+        .iter()
+        .enumerate()
+        .for_each(|(index, train_details)| {
+            train_details.as_ref().map(|train_details| {
+                let hash = train_details.compute_projection_hash_with_versioning(
+                    infra.id,
+                    infra.version,
+                    &track_section_ranges,
+                );
+                trains_hash_values.entry(hash).or_default().push(index);
+            });
+        });
+
+    let trains_hash_values_to_vec: Vec<_> = trains_hash_values.keys().cloned().collect();
```

Comment:
I would expect this variable to be named after what it is, not how it has been transformed (`_to_vec`). Maybe it could be named `train_hashes`.

Note: Perhaps `train_hash_values` is not a good name either since it relates hash to index (so it's not just a list of values). Maybe `train_hash_to_idx`?